src/agent/debuglet.ts

```diff
@@ -318,11 +369,34 @@ export class Debuglet extends EventEmitter {
         that.scheduleRegistration_(0 /* immediately */);
         that.emit('started');
       });
-
     });
   }
 
   /**
+   * isReady returns a promise that only resolved if the last breakpoint update
+   * happend within a duration (PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS). This
+   * feature is mainly used in Google Cloud Function (GCF), as it is a
+   * serverless environment and we wanted to make sure debug agent always
+   * captures the snapshots.
+   */
+  isReady(): Promise<void> {
+    if (Date.now() < this.breakpointFetchedTimestamp +
+            PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS) {
+      return Promise.resolve();
+    } else {
+      if (this.breakpointFetched.get() !== null) {
+        return this.breakpointFetched.get() as Promise<void>;
+      }
+      this.breakpointFetched.refresh();
```

Comment:
This code here suggests that it would be nice to have `refresh` return the new promise.

Reply:
Done.

Reply:
`refresh()` just refreshed the `CachedPromise.promise` so that when calling `scheduleBreakpointFetch` afterwards, that promise can be resolved. we won't resolve staled promised here. However, we will return promise after `scheduleBreakpointFetch` so will need to call `get` in the end. I guess refresh can return the newly created promise but not necessary.